{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"ReauthorizationNetworkErrorHandler"},{"kind":"text","text":" : "},{"kind":"typeIdentifier","identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkErrorHandler","preciseIdentifier":"s:10Networking19NetworkErrorHandlerP","text":"NetworkErrorHandler"}],"languages":["swift"],"platforms":["macOS"]}]},{"kind":"content","content":[{"anchor":"overview","level":2,"type":"heading","text":"Overview"},{"type":"paragraph","inlineContent":[{"type":"text","text":"This protocol extends the functionality required by a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkErrorHandler"},{"type":"text","text":" to decide whether reauthentication should be attempted by a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizingNetworkController"},{"type":"text","text":" using its "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizationProvider"},{"type":"text","text":"."}]},{"type":"paragraph","inlineContent":[{"type":"text","text":"For discussion on the "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkErrorHandler\/map(_:from:)"},{"type":"text","text":" function required by this type, refer to the "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkErrorHandler"},{"type":"text","text":" documentation."}]},{"type":"paragraph","inlineContent":[{"type":"text","text":"A "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizationProvider"},{"type":"text","text":" needs to know when a thrown error is recoverable by reauthorizing and retrying the request. That decision is usually specific to the API being used and the error types it provides. The "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizationNetworkErrorHandler\/shouldAttemptReauthorization(afterCatching:from:)"},{"type":"text","text":" function is called by a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizingNetworkController"},{"type":"text","text":", and should examine the provided error and response "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkResponse\/content"},{"type":"text","text":" to see if the error is one that can be recovered by reauthorizing."}]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/networking\/reauthorizationnetworkerrorhandler"],"traits":[{"interfaceLanguage":"swift"}]}],"relationshipsSections":[{"identifiers":["doc:\/\/Networking\/documentation\/Networking\/NetworkErrorHandler"],"kind":"relationships","title":"Inherits From","type":"inheritsFrom"}],"identifier":{"url":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizationNetworkErrorHandler","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"Describes a type that handles errors produced by a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizingNetworkController"},{"type":"text","text":"."}],"kind":"symbol","metadata":{"fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"ReauthorizationNetworkErrorHandler"}],"title":"ReauthorizationNetworkErrorHandler","roleHeading":"Protocol","role":"symbol","symbolKind":"protocol","externalID":"s:10Networking34ReauthorizationNetworkErrorHandlerP","modules":[{"name":"Networking"}],"navigatorTitle":[{"kind":"identifier","text":"ReauthorizationNetworkErrorHandler"}]},"hierarchy":{"paths":[["doc:\/\/Networking\/documentation\/Networking"]]},"topicSections":[{"title":"Instance Methods","identifiers":["doc:\/\/Networking\/documentation\/Networking\/ReauthorizationNetworkErrorHandler\/shouldAttemptReauthorization(afterCatching:from:)"]}],"references":{"doc://Networking/documentation/Networking/NetworkResponse/content":{"role":"symbol","title":"content","fragments":[{"kind":"keyword","text":"let"},{"kind":"text","text":" "},{"kind":"identifier","text":"content"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Content"}],"abstract":[{"type":"text","text":"The generic contents of the response."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkResponse\/content","kind":"symbol","type":"topic","url":"\/documentation\/networking\/networkresponse\/content"},"doc://Networking/documentation/Networking/AuthorizationProvider":{"role":"symbol","title":"AuthorizationProvider","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"AuthorizationProvider"}],"abstract":[{"type":"text","text":"Provides authorization functionality for a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkController"},{"type":"text","text":". Specifically the "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/AuthorizingNetworkController"},{"type":"text","text":"."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/AuthorizationProvider","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"AuthorizationProvider"}],"url":"\/documentation\/networking\/authorizationprovider"},"doc://Networking/documentation/Networking":{"role":"collection","title":"Networking","abstract":[],"identifier":"doc:\/\/Networking\/documentation\/Networking","kind":"symbol","type":"topic","url":"\/documentation\/networking"},"doc://Networking/documentation/Networking/ReauthorizationProvider":{"role":"symbol","title":"ReauthorizationProvider","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"ReauthorizationProvider"}],"abstract":[{"type":"text","text":"Extends the authorizing capabilities of a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/AuthorizationProvider"},{"type":"text","text":" with functions to facilitate reauthorization when needed."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizationProvider","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"ReauthorizationProvider"}],"url":"\/documentation\/networking\/reauthorizationprovider"},"doc://Networking/documentation/Networking/ReauthorizationNetworkErrorHandler":{"role":"symbol","title":"ReauthorizationNetworkErrorHandler","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"ReauthorizationNetworkErrorHandler"}],"abstract":[{"type":"text","text":"Describes a type that handles errors produced by a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizingNetworkController"},{"type":"text","text":"."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizationNetworkErrorHandler","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"ReauthorizationNetworkErrorHandler"}],"url":"\/documentation\/networking\/reauthorizationnetworkerrorhandler"},"doc://Networking/documentation/Networking/NetworkErrorHandler":{"role":"symbol","title":"NetworkErrorHandler","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"NetworkErrorHandler"}],"abstract":[{"type":"text","text":"Describes a type that handles errors produced by a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkController"},{"type":"text","text":"."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkErrorHandler","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"NetworkErrorHandler"}],"url":"\/documentation\/networking\/networkerrorhandler"},"doc://Networking/documentation/Networking/ReauthorizingNetworkController":{"role":"symbol","title":"ReauthorizingNetworkController","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"ReauthorizingNetworkController"}],"abstract":[{"type":"text","text":"A "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkController"},{"type":"text","text":" that authorizes every request submitted using the provided "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizationProvider"},{"type":"text","text":", and attempts to reauthorize the app whenever authorization fails."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizingNetworkController","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"ReauthorizingNetworkController"}],"url":"\/documentation\/networking\/reauthorizingnetworkcontroller"},"doc://Networking/documentation/Networking/NetworkResponse":{"role":"symbol","title":"NetworkResponse","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"NetworkResponse"}],"abstract":[{"type":"text","text":"Contains the response for a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkRequest"},{"type":"text","text":", wrapping some generic "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkResponse\/content"},{"type":"text","text":", a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkResponse\/statusCode"},{"type":"text","text":", and the "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkResponse\/headers"},{"type":"text","text":"."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkResponse","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"NetworkResponse"}],"url":"\/documentation\/networking\/networkresponse"},"doc://Networking/documentation/Networking/NetworkController":{"role":"symbol","title":"NetworkController","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"NetworkController"}],"abstract":[{"type":"text","text":"Defines an entrypoint for fetching content from a "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkRequest"},{"type":"text","text":"."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkController","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"NetworkController"}],"url":"\/documentation\/networking\/networkcontroller"},"doc://Networking/documentation/Networking/ReauthorizationNetworkErrorHandler/shouldAttemptReauthorization(afterCatching:from:)":{"role":"symbol","title":"shouldAttemptReauthorization(afterCatching:from:)","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"shouldAttemptReauthorization"},{"kind":"text","text":"("},{"kind":"externalParam","text":"afterCatching"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":", "},{"kind":"externalParam","text":"from"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"NetworkResponse","preciseIdentifier":"s:10Networking15NetworkResponseV"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Data","preciseIdentifier":"s:10Foundation4DataV"},{"kind":"text","text":">) -> "},{"kind":"typeIdentifier","text":"Bool","preciseIdentifier":"s:Sb"}],"abstract":[{"type":"text","text":"Decides whether or not the provided "},{"type":"codeVoice","code":"error"},{"type":"text","text":" is recoverable by reauthorizing and retrying a request. The "},{"type":"codeVoice","code":"response"},{"type":"text","text":" provided is the "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkResponse"},{"type":"text","text":" returned by the failed request that contains the raw "},{"type":"codeVoice","code":"Data"},{"type":"text","text":" for the response."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/ReauthorizationNetworkErrorHandler\/shouldAttemptReauthorization(afterCatching:from:)","kind":"symbol","required":true,"type":"topic","url":"\/documentation\/networking\/reauthorizationnetworkerrorhandler\/shouldattemptreauthorization(aftercatching:from:)"},"doc://Networking/documentation/Networking/NetworkErrorHandler/map(_:from:)":{"role":"symbol","title":"map(_:from:)","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"map"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":", "},{"kind":"externalParam","text":"from"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"NetworkResponse","preciseIdentifier":"s:10Networking15NetworkResponseV"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Data","preciseIdentifier":"s:10Foundation4DataV"},{"kind":"text","text":">) -> "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"}],"abstract":[{"type":"text","text":"Attempts to map an "},{"type":"codeVoice","code":"Error"},{"type":"text","text":" and the "},{"type":"reference","isActive":true,"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkResponse"},{"type":"text","text":" that caused it to be thrown into a more detailed error."}],"identifier":"doc:\/\/Networking\/documentation\/Networking\/NetworkErrorHandler\/map(_:from:)","kind":"symbol","required":true,"type":"topic","url":"\/documentation\/networking\/networkerrorhandler\/map(_:from:)"}}}